python/ray/tune/ray_trial_executor.py

```diff
@@ -112,9 +124,13 @@ def _stop_trial(self, trial, error=False, error_msg=None,
         try:
             trial.write_error_log(error_msg)
             if hasattr(trial, 'runner') and trial.runner:
-                stop_tasks = []
-                stop_tasks.append(trial.runner.stop.remote())
-                stop_tasks.append(trial.runner.__ray_terminate__.remote())
+                if not error and self._cached_runner is None:
+                    logger.debug("Retaining trial runner {}".format(
+                        trial.runner))
+                    self._cached_runner = trial.runner
```

Comment:
I suspect GPU trials may run into issues when reusing same processes because TF doesn't give up the GPU -- unless you've observed otherwise?

Reply:
There's no difference between reusing a runner, that retains control of the GPU, and stopping the runner and starting a new one (release / reacquire the GPU), right?

Reply:
oh I guess my question is mainly what happens during `Trainable._setup()`, which actually isn't called in `_setup_runner` when we're using a cached runner (see above note)

Reply:
Right, we call reset_trial() instead. I guess this works for PBT, but not necesssarily for other algorithms. Unless we add a reset_state() function as well?